redbot/cogs/mod/events.py

```diff
@@ -19,20 +20,30 @@ class Events(MixinMeta):
 
     async def check_duplicates(self, message):
         guild = message.guild
-        author = message.author
+        author = message.author.id
+        channel = message.channel
 
-        guild_cache = self.cache.get(guild.id, None)
-        if guild_cache is None:
-            repeats = await self.config.guild(guild).delete_repeats()
+        repeats = await self.config.guild(guild).delete_repeats()
+        repeats_channels = await self.config.guild(guild).delete_repeats_channels.all()
+        if str(channel.id) in repeats_channels:
+            if repeats_channels[str(channel.id)] == -1:
+                return False
+        else:
             if repeats == -1:
                 return False
-            guild_cache = self.cache[guild.id] = defaultdict(lambda: deque(maxlen=repeats))
 
+        guild_cache = self.cache.get(guild.id, None)
+        if guild_cache is None:
+
+            # Cache is Map (channel -> Map (author -> messages))
+            guild_cache = self.cache[guild.id] = await create_new_cache(self.config, guild)
         if not message.content:
             return False
```

Comment:
This is not a fault of your PR, but while you are touching this function it is more efficient if this is the first thing checked instead of it being here.